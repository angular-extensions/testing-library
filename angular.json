{
  "version": 1,
  "cli": {
    "analytics": false
  },
  "defaultProject": "example-app",
  "projects": {
    "example-app": {
      "projectType": "application",
      "root": "apps/example-app",
      "sourceRoot": "apps/example-app/src",
      "prefix": "app",
      "schematics": {},
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "aot": true,
            "outputPath": "dist/apps/example-app",
            "index": "apps/example-app/src/index.html",
            "main": "apps/example-app/src/main.ts",
            "polyfills": "apps/example-app/src/polyfills.ts",
            "tsConfig": "apps/example-app/tsconfig.app.json",
            "assets": ["apps/example-app/src/favicon.ico", "apps/example-app/src/assets"],
            "styles": ["apps/example-app/src/styles.css"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "apps/example-app/src/environments/environment.ts",
                  "with": "apps/example-app/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true
            }
          },
          "outputs": ["{options.outputPath}"]
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "example-app:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "example-app:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "example-app:build"
          }
        },
        "lint": {
          "builder": "@nrwl/linter:eslint",
          "options": {
            "lintFilePatterns": ["apps/example-app/src/**/*.ts", "apps/example-app/src/**/*.html"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/example-app/jest.config.js",
            "setupFile": "apps/example-app/src/test-setup.ts"
          },
          "outputs": ["coverage/"]
        }
      }
    },
    "testing-library": {
      "root": "projects/testing-library",
      "sourceRoot": "projects/testing-library/src",
      "projectType": "library",
      "prefix": "lib",
      "architect": {
        "build-package": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "tsConfig": "projects/testing-library/tsconfig.lib.json",
            "project": "projects/testing-library/ng-package.json"
          },
          "configurations": {
            "production": {
              "project": "projects/testing-library/ng-package.json",
              "tsConfig": "projects/testing-library/tsconfig.lib.json"
            }
          }
        },
        "lint": {
          "builder": "@nrwl/linter:eslint",
          "options": {
            "lintFilePatterns": [
              "projects/testing-library/src/**/*.ts",
              "projects/testing-library/src/**/*.html",
              "projects/testing-library/src/**/*.html"
            ]
          }
        },
        "build": {
          "builder": "@nrwl/workspace:run-commands",
          "options": {
            "parallel": false,
            "commands": [
              {
                "command": "ng run testing-library:build-package"
              },
              {
                "command": "cpy ./README.md ./dist/@testing-library/angular"
              }
            ]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "projects/testing-library/jest.config.js",
            "setupFile": "projects/testing-library/test-setup.ts"
          },
          "outputs": ["coverage/projects/testing-library"]
        }
      }
    },
    "jest-utils": {
      "root": "projects/jest-utils",
      "sourceRoot": "projects/jest-utils/src",
      "projectType": "library",
      "prefix": "lib",
      "architect": {
        "build-package": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "tsConfig": "projects/jest-utils/tsconfig.lib.json",
            "project": "projects/jest-utils/ng-package.json"
          },
          "configurations": {
            "production": {
              "project": "projects/jest-utils/ng-package.json",
              "tsConfig": "projects/jest-utils/tsconfig.lib.json"
            }
          }
        },
        "lint": {
          "builder": "@nrwl/linter:eslint",
          "options": {
            "lintFilePatterns": [
              "projects/jest-utils/src/**/*.ts",
              "projects/jest-utils/src/**/*.html",
              "projects/jest-utils/src/**/*.html"
            ]
          }
        },
        "build": {
          "builder": "@nrwl/workspace:run-commands",
          "options": {
            "parallel": false,
            "commands": [
              {
                "command": "ng run jest-utils:build-package"
              }
            ]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "projects/jest-utils/jest.config.js",
            "setupFile": "projects/jest-utils/test-setup.ts"
          },
          "outputs": ["coverage/projects/jest-utils"]
        }
      }
    }
  }
}
