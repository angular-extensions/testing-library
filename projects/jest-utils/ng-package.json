{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../@angular-extensions/testing-library/jest-utils",
  "deleteDestPath": false,
  "lib": {
    "entryFile": "src/public_api.ts"
  }
}
