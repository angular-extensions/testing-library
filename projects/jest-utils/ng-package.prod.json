{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/@angular-extensions/testing-library/jest-utils",
  "deleteDestPath": false,
  "lib": {
    "entryFile": "src/public_api.ts"
  }
}
