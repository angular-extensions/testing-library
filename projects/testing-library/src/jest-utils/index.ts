export * from './configure-test-suite';
export * from './create-mock';
