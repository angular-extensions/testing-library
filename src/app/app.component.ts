import { Component } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { Store } from '@ngrx/store';
import { GreetService } from './greet.service';
import { FormBuilder, Validators } from '@angular/forms';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  animations: [
    trigger('openClose', [
      state(
        'open',
        style({
          height: '200px',
          opacity: 1,
          backgroundColor: 'yellow',
        }),
      ),
      state(
        'closed',
        style({
          height: '100px',
          opacity: 0.5,
          backgroundColor: 'green',
        }),
      ),
      transition('open => closed', [animate('1s')]),
      transition('closed => open', [animate('0.5s')]),
    ]),
  ],
})
export class AppComponent {
  isOpen = true;
  title = 'app';

  colors = ['red', 'blue', 'yellow', 'pink'];
  animals = [
    { name: 'Dog', sound: 'Woof!' },
    { name: 'Cat', sound: 'Meow!' },
    { name: 'Cow', sound: 'Moo!' },
    { name: 'Fox', sound: 'Wa-pa-pa-pa-pa-pa-pow!' },
  ];

  form = this.fb.group({
    name: ['', [Validators.required, Validators.minLength(2)]],
    age: ['', [Validators.min(18), Validators.max(28)]],
    color: [],
    animal: [],
    car: [],
  });

  constructor(private store: Store<any>, private greetService: GreetService, private fb: FormBuilder) {}

  greet() {
    this.greetService.greet();
  }

  onSubmit() {
    console.log('Form submitted: ', this.form.value);
  }

  toggle() {
    this.isOpen = !this.isOpen;
  }
}
