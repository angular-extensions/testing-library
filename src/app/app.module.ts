import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';

import { AppComponent } from './app.component';
import { MaterialModule } from './material.mode';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, ReactiveFormsModule, BrowserAnimationsModule, StoreModule.forRoot({}), MaterialModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
