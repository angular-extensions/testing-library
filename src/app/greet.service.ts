import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class GreetService {
  greet(): void {
    console.log('ðŸ‘‹ðŸ‘‹');
  }
}
