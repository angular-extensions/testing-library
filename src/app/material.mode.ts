import { NgModule } from '@angular/core';
import { A11yModule } from '@angular/cdk/a11y';
import { MatInputModule } from '@angular/material/input';
import { MatSelectModule } from '@angular/material/select';

@NgModule({
  exports: [A11yModule, MatInputModule, MatSelectModule],
})
export class MaterialModule {}
