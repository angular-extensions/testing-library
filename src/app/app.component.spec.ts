import { TestBed } from '@angular/core/testing';
import { Store } from '@ngrx/store';
import { provideMockStore } from '@ngrx/store/testing';

import { render } from '@testing-library/angular';
import { provideMock } from '@testing-library/angular/jest-utils';

import { AppComponent } from './app.component';
import { GreetService } from './greet.service';
import { ReactiveFormsModule } from '@angular/forms';
import { MaterialModule } from './material.mode';

test(`matches snapshot`, async () => {
  const { container } = await render('<app-root></app-root>', {
    declarations: [AppComponent],
    imports: [ReactiveFormsModule, MaterialModule],
    providers: [provideMockStore()],
  });
  expect(container).toMatchSnapshot();
});

test(`should have a title`, async () => {
  const { getByText } = await render('<app-root></app-root>', {
    declarations: [AppComponent],
    imports: [ReactiveFormsModule, MaterialModule],
    providers: [provideMockStore()],
  });
  expect(getByText('Welcome to app!')).toBeDefined();
});

test(`should render title in a h1 tag`, async () => {
  const { container } = await render('<app-root></app-root>', {
    declarations: [AppComponent],
    imports: [ReactiveFormsModule, MaterialModule],
    providers: [provideMockStore()],
  });
  expect(container.querySelector('h1').textContent).toContain('Welcome to app!');
});

test(`should be able to get the Store`, async () => {
  await render('<app-root></app-root>', {
    declarations: [AppComponent],
    imports: [ReactiveFormsModule, MaterialModule],
    providers: [provideMockStore()],
  });
  expect(TestBed.get<Store<any>>(Store)).toBeDefined();
});

test(`should provide a mock greet service`, async () => {
  const component = await render(AppComponent, {
    declarations: [AppComponent],
    imports: [ReactiveFormsModule, MaterialModule],
    providers: [provideMockStore(), provideMock(GreetService)],
  });
  const service: GreetService = TestBed.get<GreetService>(GreetService);

  component.click(component.getByText('Greet'));

  expect(service.greet).toHaveBeenCalled();
});

describe('Forms', () => {
  test(`should have form validations`, async () => {
    const component = await render(AppComponent, {
      imports: [ReactiveFormsModule, MaterialModule],
      providers: [provideMockStore()],
    });

    const appComponent = component.fixture.componentInstance as AppComponent;
    expect(appComponent.form.valid).toBe(false);

    const nameInput = component.getByLabelText('Name:');
    const ageInput = component.getByLabelText('Age:');
    const colorInput = component.getByLabelText('Favorite color:');
    const animalInput = component.getByLabelText('Favorite animal:');
    const carInput = component.getByLabelText(/car/);

    const nameValue = appComponent.form.get('name');
    const ageValue = appComponent.form.get('age');

    component.type(nameInput, 'B');
    expect(nameValue.valid).toBe(false);

    component.type(nameInput, 'Bob');
    expect(nameValue.valid).toBe(true);

    component.type(ageInput, '17');
    expect(ageValue.valid).toBe(false);

    component.type(ageInput, '61');
    expect(ageValue.valid).toBe(false);

    component.type(ageInput, '20');
    expect(ageValue.valid).toBe(true);

    component.selectOptions(colorInput, 'ink', { exact: false });
    component.selectOptions(colorInput, /YELLOW/i);
    component.selectOptions(animalInput, 'Cow');
    component.selectOptions(carInput, 'Audi');

    expect(appComponent.form.valid).toBe(true);
    expect(appComponent.form.value).toEqual({
      name: 'Bob',
      age: 20,
      color: 'yellow',
      animal: { name: 'Cow', sound: 'Moo!' },
      car: 'audi',
    });
  });
});
