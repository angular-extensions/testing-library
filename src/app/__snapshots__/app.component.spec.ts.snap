// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="root0"
>
  <app-root
    class="ng-tns-c0-0"
  >
    <div
      class="ng-tns-c0-0"
      style="text-align:center"
    >
      <h1
        class="ng-tns-c0-0"
      >
        Welcome to app!
      </h1>
      <img
        alt="Angular Logo"
        class="ng-tns-c0-0"
        src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNTAgMjUwIj4KICAgIDxwYXRoIGZpbGw9IiNERDAwMzEiIGQ9Ik0xMjUgMzBMMzEuOSA2My4ybDE0LjIgMTIzLjFMMTI1IDIzMGw3OC45LTQzLjcgMTQuMi0xMjMuMXoiIC8+CiAgICA8cGF0aCBmaWxsPSIjQzMwMDJGIiBkPSJNMTI1IDMwdjIyLjItLjFWMjMwbDc4LjktNDMuNyAxNC4yLTEyMy4xTDEyNSAzMHoiIC8+CiAgICA8cGF0aCAgZmlsbD0iI0ZGRkZGRiIgZD0iTTEyNSA1Mi4xTDY2LjggMTgyLjZoMjEuN2wxMS43LTI5LjJoNDkuNGwxMS43IDI5LjJIMTgzTDEyNSA1Mi4xem0xNyA4My4zaC0zNGwxNy00MC45IDE3IDQwLjl6IiAvPgogIDwvc3ZnPg=="
        width="300"
      />
    </div>
    <h2
      class="ng-tns-c0-0"
    >
      Here are some links to help you start:
    </h2>
    <ul
      class="ng-tns-c0-0"
    >
      <li
        class="ng-tns-c0-0"
      >
        <h2
          class="ng-tns-c0-0"
        >
          <a
            class="ng-tns-c0-0"
            href="https://angular.io/tutorial"
            rel="noopener"
            target="_blank"
          >
            Tour of Heroes
          </a>
        </h2>
      </li>
      <li
        class="ng-tns-c0-0"
      >
        <h2
          class="ng-tns-c0-0"
        >
          <a
            class="ng-tns-c0-0"
            href="https://github.com/angular/angular-cli/wiki"
            rel="noopener"
            target="_blank"
          >
            CLI Documentation
          </a>
        </h2>
      </li>
      <li
        class="ng-tns-c0-0"
      >
        <h2
          class="ng-tns-c0-0"
        >
          <a
            class="ng-tns-c0-0"
            href="https://blog.angular.io/"
            rel="noopener"
            target="_blank"
          >
            Angular blog
          </a>
        </h2>
      </li>
    </ul>
    <button
      class="ng-tns-c0-0"
    >
      Greet
    </button>
    <form
      class="ng-tns-c0-0 ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <label
        class="ng-tns-c0-0"
      >
         Name: 
        <input
          class="ng-tns-c0-0 ng-untouched ng-pristine ng-invalid"
          formcontrolname="name"
          ng-reflect-name="name"
          type="text"
        />
      </label>
      <label
        class="ng-tns-c0-0"
      >
         Age: 
        <input
          class="ng-tns-c0-0 ng-untouched ng-pristine ng-valid"
          formcontrolname="age"
          ng-reflect-name="age"
          type="number"
        />
      </label>
      <label
        class="ng-tns-c0-0"
        for="color"
      >
        Favorite color:
      </label>
      <select
        class="ng-tns-c0-0 ng-untouched ng-pristine ng-valid"
        formcontrolname="color"
        id="color"
        ng-reflect-name="color"
      >
        
        <option
          class="ng-tns-c0-0 ng-star-inserted"
          ng-reflect-ng-value="red"
          value="0: red"
        >
          red
        </option>
        <option
          class="ng-tns-c0-0 ng-star-inserted"
          ng-reflect-ng-value="blue"
          value="1: blue"
        >
          blue
        </option>
        <option
          class="ng-tns-c0-0 ng-star-inserted"
          ng-reflect-ng-value="yellow"
          value="2: yellow"
        >
          yellow
        </option>
        <option
          class="ng-tns-c0-0 ng-star-inserted"
          ng-reflect-ng-value="pink"
          value="3: pink"
        >
          pink
        </option>
      </select>
      <h4
        class="ng-tns-c0-0"
      >
        mat select
      </h4>
      <mat-form-field
        class="mat-form-field ng-tns-c1-1 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-infix"
            >
              <mat-select
                aria-disabled="false"
                aria-invalid="false"
                aria-labelledby="mat-form-field-label-1"
                aria-multiselectable="false"
                aria-required="false"
                class="mat-select ng-tns-c2-2 ng-tns-c0-0 ng-untouched ng-pristine ng-valid mat-select-empty ng-star-inserted"
                formcontrolname="animal"
                id="mat-select-0"
                ng-reflect-name="animal"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-hidden="true"
                  cdk-overlay-origin=""
                  class="mat-select-trigger"
                >
                  <div
                    class="mat-select-value"
                    ng-reflect-ng-switch="true"
                  >
                    
                    <span
                      class="mat-select-placeholder ng-tns-c2-2 ng-star-inserted"
                    >
                      Â 
                    </span>
                    
                  </div>
                  <div
                    class="mat-select-arrow-wrapper"
                  >
                    <div
                      class="mat-select-arrow"
                    />
                  </div>
                </div>
                
              </mat-select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  aria-owns="mat-select-0"
                  class="mat-form-field-label ng-tns-c1-1 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-select-0"
                  id="mat-form-field-label-1"
                  ng-reflect-disabled="true"
                  ng-reflect-ng-switch="true"
                >
                  
                  
                  <mat-label
                    class="ng-tns-c0-0 ng-star-inserted"
                  >
                    Favorite animal:
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-underline ng-tns-c1-1 ng-star-inserted"
          >
            <span
              class="mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-tns-c1-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
      <h4
        class="ng-tns-c0-0"
      >
        native html select
      </h4>
      <mat-form-field
        class="mat-form-field ng-tns-c1-3 mat-primary mat-form-field-type-mat-native-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
      >
        <div
          class="mat-form-field-wrapper"
        >
          <div
            class="mat-form-field-flex"
          >
            
            
            <div
              class="mat-form-field-infix"
            >
              <select
                aria-invalid="false"
                aria-required="false"
                class="mat-input-element mat-form-field-autofill-control ng-tns-c0-0 cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
                formcontrolname="car"
                id="mat-input-0"
                matnativecontrol=""
                ng-reflect-name="car"
              >
                <option
                  class="ng-tns-c0-0"
                  label="--select something --"
                />
                <option
                  class="ng-tns-c0-0"
                  ng-reflect-value="saab"
                  value="saab"
                >
                  Saab
                </option>
                <option
                  class="ng-tns-c0-0"
                  ng-reflect-value="mercedes"
                  value="mercedes"
                >
                  Mercedes
                </option>
                <option
                  class="ng-tns-c0-0"
                  ng-reflect-value="audi"
                  value="audi"
                >
                  Audi
                </option>
              </select>
              <span
                class="mat-form-field-label-wrapper"
              >
                
                <label
                  aria-owns="mat-input-0"
                  class="mat-form-field-label ng-tns-c1-3 mat-empty mat-form-field-empty ng-star-inserted"
                  for="mat-input-0"
                  id="mat-form-field-label-3"
                  ng-reflect-disabled="true"
                  ng-reflect-ng-switch="true"
                >
                  
                  
                  <mat-label
                    class="ng-tns-c0-0 ng-star-inserted"
                  >
                    Select your car:
                  </mat-label>
                  
                  
                </label>
              </span>
            </div>
            
          </div>
          
          <div
            class="mat-form-field-underline ng-tns-c1-3 ng-star-inserted"
          >
            <span
              class="mat-form-field-ripple"
            />
          </div>
          <div
            class="mat-form-field-subscript-wrapper"
            ng-reflect-ng-switch="hint"
          >
            
            
            <div
              class="mat-form-field-hint-wrapper ng-tns-c1-3 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
              style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
            >
              
              <div
                class="mat-form-field-hint-spacer"
              />
            </div>
          </div>
        </div>
      </mat-form-field>
    </form>
    <div
      class="ng-tns-c0-0"
    >
      <button
        class="ng-tns-c0-0"
      >
        Toggle Open/Close
      </button>
      <div
        class="open-close-container ng-trigger ng-trigger-openClose"
        style="height:200px;opacity:1;background-color:yellow;0:height;1:opacity;2:background-color;"
      >
        <p
          class="ng-tns-c0-0"
        >
          The box is now Open!
        </p>
      </div>
    </div>
  </app-root>
</div>
`;
